### PR TITLE
cpio: use SectionReaders, not LimitedReaders

### PR DESCRIPTION
**Pull request by rminnich**
LimitedReaders move the offset, and we want readers that
do not do that, since Read returns Files with Readers in
them.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
oh darn I added another bit here by mistake. take a look.

xxd -include the-cpio-file.cpio

then edit by hand.

SectionReaders are *really* handy.

File: cmds/cpio/fs_unix.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+func create(f *File) error {
+
+	m, err := cpioModetoMode(f.Mode)
+	if err != nil {
+		return err
+	}
+
+	switch m {
+	case os.ModeSocket:
```
Comment (rjoleary) [inline cmds/cpio/fs_unix.go]: Go allows commas in switches, so you can go:

```
case os.ModeSocket, os.ModeDevice, os.ModeCharDevice:
    return fmt.Errorf(...)
```

And combine a few of these cases into 1.

Comment (rminnich) [inline cmds/cpio/fs_unix.go]: in this case that would not last long, these are going to get filled in next.

Comment (rminnich) [inline cmds/cpio/fs_unix.go]: as for the binary blob, I'll look at the go tree again, but my impression is that they tend to prefer initialized arrays (like this one) to binary blobs.
thanks.

**Pull request merged**